### PR TITLE
Update supported .NET Versions docs

### LINKED ISSUE #2138: Remove frameworks older than net461
DiagnosticSource 6.0 only supports net461 and above. (i.e it drops net452,net46 support, few months before they are officially going unsupported - https://devblogs.microsoft.com/dotnet/net-framework-4-5-2-4-6-4-6-1-will-reach-end-of-support-on-april-26-2022/) Since the API/SDK/Exporters depend on DiagnosticSource , they can now only support net461 and above.

Proposing to drop support for NET452, NET46 in the next release (1.2.0). What this means is, when we release the next minor version (1.2.0) in Nov 2021, we'll be cutting support for some frameworks (net452, net46 specifically). This might be considered a breaking-change, and warrant a major version bump.

However, I think we can (and we should) do minor version bump only, considering NET452/46 support is dropped by DS (which we have hard dependency on), and Net452/net46 will be EOL within 4 months of our planned 1.2 release.

@open-telemetry/dotnet-approvers @open-telemetry/dotnet-maintainers Please share your concerns if any with this. Otherwise, please mark approval by commenting to the issue.

### PR DESCRIPTION
**Pull request by pellared**
Part of #2138.

## Changes

Update supported .NET Versions docs.

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #2138
* [ ] Changes in public API reviewed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,8 +16,8 @@ Packages shipped from this repository generally support all the officially
 supported versions of [.NET
 Core](https://dotnet.microsoft.com/download/dotnet-core), and [.NET
 Framework](https://dotnet.microsoft.com/download/dotnet-framework) except for
-`.NET Framework 3.5 SP1`. Any exceptions to this are noted in the individual
-`README.md` files.
+versions lower than `.NET Framework 4.6.1`.
```
Comment (reyang) [inline README.md:19]: It seems we don't support `netcoreapp2.1`, too.

Comment (pellared) [reply README.md:19]: I see `netstandard2.0` in at least `opentelemetry.1.2.0-alpha1.nupkg` which supports `.NET Core 2.0+`.
See: https://docs.microsoft.com/en-us/dotnet/standard/net-standard

Comment (cijothomas) [reply README.md:19]: If we are calling out .NET Core 2.1, we need to remove that on Aug 22 when .NET Core 2.1 goes out of support (so the original statement "support all the officially supported version of .NET Core" will cover it).

**Review by cijothomas — APPROVED** [review]
NEt461 part is correct. Will decide if we want to call out .net core 2.1 separately.

**Review by reyang — APPROVED** [review]

**Pull request merged**